pytorch_lightning/trainer/trainer.py

```diff
@@ -710,6 +707,60 @@ def run_sanity_check(self, ref_model):
             self.on_sanity_check_end()
             self.running_sanity_check = False
 
+    def validate(
+        self,
+        model: Optional[LightningModule] = None,
+        val_dataloaders: Optional[Union[DataLoader, List[DataLoader]]] = None,
+        ckpt_path: Optional[str] = 'best',
+        verbose: bool = True,
+        datamodule: Optional[LightningDataModule] = None,
+    ):
+        r"""
+
+        Perform one evaluation epoch over the validation set.
+
+        Args:
+            ckpt_path: Either ``best`` or path to the checkpoint you wish to evaluate.
+                If ``None``, use the current weights of the model. Default to ``best``.
+
+            datamodule: A instance of :class:`LightningDataModule`.
+
+            model: The model to evaluate.
+
+            val_dataloaders: Either a single PyTorch DataLoader or a list of them,
+                specifying validation samples.
+
+            verbose: If True, prints the evaluation results
+
+        Returns:
+            The final validation results dictionary. If no validation_epoch_end
+            is defined, returns a list of dictionaries.
+        """
```

Comment:
```suggestion
        Perform one evaluation epoch over the validation set.

        Args:
            ckpt_path: Either ``best`` or path to the checkpoint you wish to validate.
                If ``None``, use the current weights of the model. Default to ``best``.
            datamodule: A instance of :class:`LightningDataModule`.
            model: The model to evaluate.
            val_dataloaders: Either a single PyTorch DataLoader or a list of them,
                specifying validation samples.
            verbose: If True, prints the validation results

        Returns:
            The final validation returns a dictionary. If no validation_epoch_end
            is defined, returns a list of dictionaries.
        """
```

Reply:
Good, maybe it's worth it to do the same in `Trainer.test()` too?

Reply:
yeap

Reply:
Alright, committed in f8ab391. I also expanded the _Returns_ section a bit to clarify what's being returned.